devito/parameters.py

```diff
@@ -258,6 +258,29 @@ def wrapper(*args, **kwargs):
         return wrapper
 
 
+class switch_log_level(object):
+    """
+    A context manager subclassing `switchconfig` to temporarily change
+    MPI logging.
+    """
+
+    def __init__(self, comm):
+
+        self.level = configuration['log-level']
+        self.comm = comm
+
+        # Limit logging to rank 0
+        set_log_level(self.level, comm)
```

Comment:
theoretically shouldn't this be inside `__enter__`? though in practice it might not change much

Reply:
fixed